src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -4374,10 +4382,9 @@ private bool RemoteDirectoryExist(System.Management.Automation.PowerShell ps, st
             ps.AddParameter("dirPathExists", path);
             Hashtable op = SafeInvokeCommand.Invoke(ps, this, null);
 
-            if (op != null)
+            if (op["Exists"] != null)
```

Comment:
Please revert - we need the null check.

Reply:
Should be `if (op?["Exists"] != Null)` to cover both cases in one.